# Homework 1 Python Basics Practice

1.) Accept the user's first and last name and print them in reverse order with a space between them

In [1]:
firstname = input("Tell me your first name: ")
lastname = input("Tell me your last name: ")
print ("Hello my name is " + lastname + " " + firstname)

Tell me your first name: Luis
Tell me your last name: Pena
Hello my name is Pena Luis


2.) Accept an integer (n) and compute the value of n+nn+nnn

In [3]:
n = int(input("Give me any number: "))
nn = ((n*10) + n)
nnn = ((n*100)+(n*10)+n)
print(n + nn + nnn)

Give me any number: 3
369


3.) Ask the user "What country are you from?" then print the following statement: "I have heard that [input] is a beautiful country!

In [6]:
country = str(input("What country are you from:"))
print("I have heard that " + country + " is a beautiful country. ")

What country are you from:Mexico
I have heard that Mexico is a beautiful country. 


4.) What is the output of x AND explain why that is the output for:

x = 36 / 4 * (3 +  2) * 4 + 2

In [7]:
x = 36 / 4 * (3 + 2) * 4 + 2
print(x)

182.0


This mathematical calculation follows the mathematical order of operations PEMDAS (Parenthesis, Exponents, Multiplication, Division, Addition, Subtraction). In this case, the (3 + 2) would equal 5 as the first step. Then in this case the Multiplication and Division goes by order from left to right, so 36/4 = 9 * (5) * 4 = 180. Then lastly goes the addition/subtraction in this case 180 + 2 = 182.

5.) Is a string immutable in python?

An immutable object is an object that is not changeable and its state cannot be modified once it is created. Meaning in the case of a string once it is created that string will always represent the same value.